recipes/brpc/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+from conans import CMake, ConanFile, tools
+import glob
+import os
+
+
+class BrpcConan(ConanFile):
+    name = "brpc"
+    description = "An industrial-grade RPC framework used throughout Baidu"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apache/incubator-brpc"
+    topics = ("conan", "brpc", "baidu", "rpc")
+    license = ("Apache-2.0")
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_snappy": [True, False],
+        "with_glog": [True, False],
+        "with_thrift": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "with_snappy": True,
+        "with_glog": True,
+        "with_thrift": False,
+        "shared": False,
+        "fPIC": True,
+    }
+
+    _cmake = None
+
+    def requirements(self):
+        self.requires("gflags/2.2.2")
+        self.requires("protobuf/3.9.1")
+        self.requires("leveldb/1.22")
+        self.requires("openssl/1.1.1g")
+        if self.options.with_glog:
+            self.requires("glog/0.4.0")
+        if self.options.with_thrift:
+            self.requires("thrift/0.13.0")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        self.options['protobuf'].with_zlib = True
+        self.options['leveldb'].with_snappy = self.options.with_snappy
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("incubator-brpc-" + self.version, self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.patch(**self.conan_data["main_patches"][self.version])
+        if self.options.with_snappy:
+            tools.patch(**self.conan_data["snappy_patches"][self.version])
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BRPC_REVISION"] = self.conan_data["git_hashes"][self.version]
```

Comment:
What are you trying to do here?
Because the hooks will disallow this.
Is it required?

Reply:
Yeah I removed it from the conanfile, and moved it into the patch. brpc's cmakelists uses git to parse out the git hash - i don't remember if it is actually embedded somewhere in the binaries, but anyway i moved it out.